sdk/cosmosdb/cosmos/src/queryExecutionContext/orderByComparator.ts

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { NonStreamingOrderByResult } from "./nonStreamingOrderByResult";
+
+/** @hidden */
+const TYPEORDCOMPARATOR: {
+  [type: string]: { ord: number; compFunc?: (a: any, b: any) => number };
+} = Object.freeze({
+  NoValue: {
+    ord: 0,
+  },
+  undefined: {
+    ord: 1,
+  },
+  boolean: {
+    ord: 2,
+    compFunc: (a: boolean, b: boolean) => {
+      return a === b ? 0 : a > b ? 1 : -1;
+    },
+  },
+  number: {
+    ord: 4,
+    compFunc: (a: number, b: number) => {
+      return a === b ? 0 : a > b ? 1 : -1;
+    },
+  },
+  string: {
+    ord: 5,
+    compFunc: (a: string, b: string) => {
+      return a === b ? 0 : a > b ? 1 : -1;
+    },
+  },
+});
+
+/** @hidden */
+export class OrderByComparator {
+  constructor(public sortOrder: string[]) {}
+
+  public compareItems(item1: NonStreamingOrderByResult, item2: NonStreamingOrderByResult): number {
+    const orderByItemsRes1 = this.getOrderByItems(item1);
+    const orderByItemsRes2 = this.getOrderByItems(item2);
+
+    for (let i = 0; i < orderByItemsRes1.length; i++) {
+      // compares the orderby items one by one
+      const compRes = this.compareOrderByItem(orderByItemsRes1[i], orderByItemsRes2[i]);
+      if (compRes !== 0) {
+        if (this.sortOrder[i] === "Descending") {
+          return compRes;
+        } else if (this.sortOrder[i] === "Ascending") {
+          return -compRes;
+        }
+      }
+    }
+  }
+
+  private getOrderByItems(res: any): any {
+    return res["orderByItems"];
+  }
+
+  private compareOrderByItem(orderByItem1: any, orderByItem2: any): number {
+    const type1 = this.getType(orderByItem1);
+    const type2 = this.getType(orderByItem2);
+    return this.compareValue(orderByItem1["item"], type1, orderByItem2["item"], type2);
+  }
+
+  private getType(
```

Comment:
There seems to be a mismatch between return types of this method and TYPEORDCOMPARATOR. Can we make this definition a bit tighter. If string, number, boolean are the ones we support let's keep them only.
Also is "bigint" not needed?

Reply:
return type currently is same as the possible values of `typeof ` operator. Changing this will throw an exception.